CMakeLists.txt

```diff
@@ -12,14 +12,25 @@ option(BUILD_SHARED_LIBS "" OFF)
 set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
 
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/user-includes.cmake")
-   include("${CMAKE_CURRENT_SOURCE_DIR}/user-includes.cmake")
+    include("${CMAKE_CURRENT_SOURCE_DIR}/user-includes.cmake")
 endif()
 
 add_subdirectory(cpr)
 add_subdirectory(minhook)
 add_subdirectory(plog)
 
-find_library(shlwapi Shlwapi.lib REQUIRED)
+if (DEFINED CMAKE_MINGW32)
+    set(CMAKE_CXX_EXTENSIONS False)
+    find_library(shlwapi libshlwapi REQUIRED)
+    set(CMAKE_EXE_COMPILER_FLAGS  "${CMAKE_EXE_COMPILER_FLAGS} -std=c++17 -pthread")
+    set(CMAKE_CXX_FLAGS "-std=c++17 ${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_STANDARD_LIBRARIES "-static-libgcc -static-libstdc++ -lwsock32 -lws2_32 ${CMAKE_CXX_STANDARD_LIBRARIES}")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-Bstatic,--whole-archive -lwinpthread -Wl,--no-whole-archive -pthread -lshlwapi")
```

Comment:
Some of these flags are redundant (e.g. `-std=c++17` should be set by the `CMAKE_CXX_STANDARD` at the top), and the rest should be specified through `target_compile_options` or similar. This is something that can wait until we're sure what all's needed though, can compare what you and I need to get it working.

Reply:
there are issues with -std=gnu++17 and I was just trying to ensure the right flags were used.

Reply:
Addressed.